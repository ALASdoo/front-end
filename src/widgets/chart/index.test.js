import { __test__ } from './index';

describe('Chart', () => {
  describe('getPopulation', () => {
    it('should handle empty', () => {
      const result = __test__.getPopulation([]);
      expect(result).toEqual([]);
    });

    it('should filter out empty population data', () => {
      const result = __test__.getPopulation([
        { year: '2012', city: 'A', population: 10 },
        { year: '2013', city: 'A', population: 15 },
        { year: '2012', city: 'B', population: null },
        { year: '2012', city: 'C' },
        { year: '2014', city: 'B', population: 0 },
      ]);
      expect(result).toMatchSnapshot();
    });
  });

  describe('getEstimatedPopulation', () => {
    it('should handle empty', () => {
      const result = __test__.getEstimatedPopulation([]);
      expect(result).toEqual([]);
    });

    it('should filter out empty population data', () => {
      const result = __test__.getEstimatedPopulation([
        { year: '2012', city: 'A', population: 10 },
        { year: '2013', city: 'A', population: 15 },
        { year: '2012', city: 'B', population: null },
        { year: '2012', city: 'C' },
        { year: '2014', city: 'B', population: 0 },
        { year: '2022', city: 'A', population: 10, projected: 20 },
        { year: '2023', city: 'A', population: 15, projected: 25 },
        { year: '2022', city: 'B', population: null, projected: null },
        { year: '2022', city: 'C' },
        { year: '2024', city: 'B', projected: 0 },
      ]);
      expect(result).toMatchSnapshot();
    });
  });
});
