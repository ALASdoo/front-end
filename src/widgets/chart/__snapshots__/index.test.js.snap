// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chart getEstimatedPopulation should filter out empty population data 1`] = `
Array [
  Array [
    "2022",
    20,
  ],
  Array [
    "2023",
    25,
  ],
  Array [
    "2024",
    0,
  ],
]
`;

exports[`Chart getPopulation should filter out empty population data 1`] = `
Array [
  Array [
    "2012",
    10,
  ],
  Array [
    "2013",
    15,
  ],
  Array [
    "2014",
    0,
  ],
]
`;
